File: 006.cpp
// difference between sum of squares of first one hundred natural numbers
// and square of sum

#include <iostream>

using std::cout;
using std::endl;

unsigned sum_squares(const unsigned limit) {
	unsigned sum = 0;
	for (unsigned i = 0; i <= limit; ++i) {
		sum += i * i;
	}
	return sum;
}

unsigned square_sum(const unsigned limit) {
	unsigned sum = 0;
	for (unsigned i = 0; i <= limit; ++i) {
		sum += i;
	}
	return sum * sum;
}

int main() {
	const unsigned limit {100};
	cout << square_sum(limit) - sum_squares(limit) << endl;

  return 0;
}
  

File: 006.py
'''Sum square difference.'''

def sum_of_squares(n):
    '''Return the sum of the squares of the first n natural numbers.'''
    return sum(x * x for x in range(n + 1))

def square_of_sum(n):
    '''Return the square of the sum of the first x natural numbers.'''
    return sum(range(n + 1)) ** 2

if __name__ == '__main__':
    print(square_of_sum(100) - sum_of_squares(100))
